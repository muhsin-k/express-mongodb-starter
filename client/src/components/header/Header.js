import React from "react";
import PropTypes from "prop-types";

function Header(props) {
  return (
    <div>
      <nav>
        <div className="nav-wrapper">
          <a href="/" className="brand-logo center">
            {props.title}
          </a>
        </div>
      </nav>
    </div>
  );
}
Header.propTypes = {
  title: PropTypes.string
};
Header.defaultProps = {
  title: "Contact Management"
};

export default Header;
