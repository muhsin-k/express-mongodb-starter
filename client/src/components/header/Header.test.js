import React from "react";
import ReactDOM from "react-dom";
import * as enzyme from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Header from "./Header";

enzyme.configure({
  adapter: new Adapter()
});

describe("Header", () => {
  it("renders without crashing[Normal]", () => {
    enzyme.mount(<Header />);
  });

  // Snapshot
  it("should render correctly[Snapshot]", () => {
    const output = enzyme.shallow(<Header title="Contact Management" />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it("renders title ", () => {
    const wrapper = enzyme.shallow(<Header />);
    const welcome = (
      <a href="/" className="brand-logo center">
        Contact Management
      </a>
    );
    expect(wrapper.contains(welcome)).toEqual(true);
  });
});
