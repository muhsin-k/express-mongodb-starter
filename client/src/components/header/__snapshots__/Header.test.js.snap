// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly[Snapshot] 1`] = `
<div>
  <nav>
    <div
      className="nav-wrapper"
    >
      <a
        className="brand-logo center"
        href="/"
      >
        Contact Management
      </a>
    </div>
  </nav>
</div>
`;
